### ISSUE TITLE
consul_catalog: incorrect service address

### ISSUE THREAD

**Issue by chulkilee** [containous/traefik#350]
Found that consul_catalog is picking up incorrect address if registrator runs with `-internal` option. In this case consul has different `Address` and `ServiceAddress`, which are respectively consul agent ip and actual service ip.

## docker-compose.yml

```yaml
consul:
  container_name: consul
  image: consul:latest
  command: consul agent -dev -dc=local -node=docker -client=0.0.0.0 -bootstrap -log-level=INFO
  ports:
    - 8500:8500
  environment:
    - SERVICE_IGNORE=true

registrator:
  container_name: registrator
  image: gliderlabs/registrator:latest
  links:
    - consul
  command: -internal consul://consul:8500
  volumes:
    - /var/run/docker.sock:/tmp/docker.sock

traefik:
  container_name: traefik
  image: traefik:latest
  links:
    - consul
  ports:
    - 80:80
    - 8000:8080
  volumes:
    - ./traefik.toml:/etc/traefik/traefik.toml:ro
  environment:
    - SERVICE_80_NAME=traefik-http
    - SERVICE_8080_NAME=traefik-dashboard
```

## traefik.toml

```toml
logLevel = "DEBUG"
defaultEntryPoints = ["http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"

[retry]
attemps = 3

[web]
address = ":8080"

[consulCatalog]
endpoint = "consul:8500"
domain = "test"
```

## consul: catalog/node/:name

`curl -s $(docker-machine ip):8500/v1/catalog/node/docker | jq`
```json
{
  "Node": {
    "Node": "docker",
    "Address": "172.17.0.3",
    "TaggedAddresses": {
      "wan": "172.17.0.3"
    },
    "CreateIndex": 3,
    "ModifyIndex": 6
  },
  "Services": {
    "ca523626ca02:traefik:80": {
      "ID": "ca523626ca02:traefik:80",
      "Service": "traefik-http",
      "Tags": null,
      "Address": "172.17.0.5",
      "Port": 80,
      "EnableTagOverride": false,
      "CreateIndex": 4,
      "ModifyIndex": 4
    },
    "ca523626ca02:traefik:8080": {
      "ID": "ca523626ca02:traefik:8080",
      "Service": "traefik-dashboard",
      "Tags": null,
      "Address": "172.17.0.5",
      "Port": 8080,
      "EnableTagOverride": false,
      "CreateIndex": 5,
      "ModifyIndex": 5
    },
    "consul": {
      "ID": "consul",
      "Service": "consul",
      "Tags": [],
      "Address": "",
      "Port": 8300,
      "EnableTagOverride": false,
      "CreateIndex": 3,
      "ModifyIndex": 6
    }
  }
}
```

## consul: agent/services

`curl -s $(docker-machine ip):8500/v1/agent/services | jq`
```json
{
  "ca523626ca02:traefik:80": {
    "ID": "ca523626ca02:traefik:80",
    "Service": "traefik-http",
    "Tags": null,
    "Address": "172.17.0.5",
    "Port": 80,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "ca523626ca02:traefik:8080": {
    "ID": "ca523626ca02:traefik:8080",
    "Service": "traefik-dashboard",
    "Tags": null,
    "Address": "172.17.0.5",
    "Port": 8080,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "consul": {
    "ID": "consul",
    "Service": "consul",
    "Tags": [],
    "Address": "",
    "Port": 8300,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  }
}
```

## consul: catalog/service/:name

`curl -s $(docker-machine ip):8500/v1/catalog/service/traefik-dashboard | jq`
```json
[
  {
    "Node": "docker",
    "Address": "172.17.0.3",
    "ServiceID": "ca523626ca02:traefik:8080",
    "ServiceName": "traefik-dashboard",
    "ServiceTags": [],
    "ServiceAddress": "172.17.0.5",
    "ServicePort": 8080,
    "ServiceEnableTagOverride": false,
    "CreateIndex": 5,
    "ModifyIndex": 5
  }
]
```

## consul: health/service/:name

`curl -s $(docker-machine ip):8500/v1/health/service/traefik-dashboard | jq`
```json
[
  {
    "Node": {
      "Node": "docker",
      "Address": "172.17.0.3",
      "TaggedAddresses": {
        "wan": "172.17.0.3"
      },
      "CreateIndex": 3,
      "ModifyIndex": 6
    },
    "Service": {
      "ID": "ca523626ca02:traefik:8080",
      "Service": "traefik-dashboard",
      "Tags": null,
      "Address": "172.17.0.5",
      "Port": 8080,
      "EnableTagOverride": false,
      "CreateIndex": 5,
      "ModifyIndex": 5
    },
    "Checks": [
      {
        "Node": "docker",
        "CheckID": "serfHealth",
        "Name": "Serf Health Status",
        "Status": "passing",
        "Notes": "",
        "Output": "Agent alive and reachable",
        "ServiceID": "",
        "ServiceName": "",
        "CreateIndex": 3,
        "ModifyIndex": 3
      }
    ]
  }
]
```

## traefik: /api (incorrect)

`curl -s $(docker-machine ip):8000/api | jq '.consul_catalog.backends["backend-traefik-dashboard"]'`
```json
{
  "servers": {
    "server-docker-8080": {
      "url": "http://172.17.0.3:8080"
    }
  },
  "loadBalancer": {
    "method": "wrr"
  }
}
```

**Comment by chulkilee** [comment]
Oh, never mind. It is fixed at #294 - could you release new docker image? The latest version is `v1.0.0-beta.404` which is pretty old.

https://hub.docker.com/r/library/traefik/

**Comment by emilevauge** [comment]
In the meantime, you can use the image containous/traefik which is up to date :)
